### PR TITLE
Remove sufia dependency

### PR DESCRIPTION
**Pull request by flyingzumwalt**
(Rebased @jpstroop's work)

### CODE REVIEW

File: curation_concerns-models/Gemfile [hunk 1]
```diff
@@ -0,0 +1,4 @@
+source 'https://rubygems.org'
+
+# Specify your gem's dependencies in sufia-models.gemspec
```
Comment (flyingzumwalt) [inline curation_concerns-models/Gemfile]: # Specify your gem's dependencies in curation_concerns-models.gemspec

File: curation_concerns-models/app/actors/curation_concerns/base_actor.rb [hunk 2]
```diff
@@ -74,9 +74,18 @@ def attach_file(generic_file, file_to_attach)
...
+        else
+          h[k] = v
+        end
       end
     end
+    # def remove_blank_attributes!
+    #   multivalued_form_attributes.each_with_object(attributes) do |(k, v), h|
+    #     h[k] = v.select(&:present?)
+    #   end
+    # end
```
Comment (flyingzumwalt) [inline curation_concerns-models/app/actors/curation_concerns/base_actor.rb]: Remove commented lines.

File: curation_concerns-models/app/actors/curation_concerns/base_actor.rb [hunk 3]
```diff
@@ -57,7 +57,7 @@ def save
     def apply_save_data_to_curation_concern
       attributes[:rights] = Array(attributes[:rights]) if attributes.key? :rights
       remove_blank_attributes!
-      curation_concern.attributes = attributes
+      curation_concern.attributes = attributes.symbolize_keys
```
Comment (flyingzumwalt) [inline curation_concerns-models/app/actors/curation_concerns/base_actor.rb]: Why did you make this change?

File: curation_concerns-models/app/actors/curation_concerns/base_actor.rb [hunk 4]
```diff
@@ -74,9 +74,18 @@ def attach_file(generic_file, file_to_attach)
     # => { 'title' => ['first', 'second'] }
     def remove_blank_attributes!
       multivalued_form_attributes.each_with_object(attributes) do |(k, v), h|
-        h[k] = v.select(&:present?)
```
Comment (flyingzumwalt) [inline curation_concerns-models/app/actors/curation_concerns/base_actor.rb]: Why did you make this change?

File: curation_concerns-models/app/actors/curation_concerns/generic_file/actor.rb [hunk 5]
```diff
@@ -0,0 +1,154 @@
+module CurationConcerns::GenericFile
+  # Actions are decoupled from controller logic so that they may be called from a controller or a background job.
+  class Actor
```
Comment (flyingzumwalt) [inline curation_concerns-models/app/actors/curation_concerns/generic_file/actor.rb]: Delete this file.  It was already merged into `curation_concerns-models/app/actors/curation_concerns/generic_file_actor.rb`

File: curation_concerns-models/app/jobs/batch_update_job.rb [hunk 6]
```diff
@@ -41,22 +42,17 @@ def update_file(gf, user)
...
-      gf.save!
-    rescue RSolr::Error::Http => error
-      save_tries += 1
-      ActiveFedora::Base.logger.warn "BatchUpdateJob caught RSOLR error on #{gf.id}: #{error.inspect}"
-      # fail for good if the tries is greater than 3
-      raise error if save_tries >=3
-      sleep 0.01
-      retry
-    end #
-    Sufia.queue.push(ContentUpdateEventJob.new(gf.id, login))
```
Comment (flyingzumwalt) [inline curation_concerns-models/app/jobs/batch_update_job.rb]: Why did you remove these lines?

File: curation_concerns-models/app/jobs/create_derivatives_job.rb [hunk 7]
```diff
@@ -1,14 +0,0 @@
-class CreateDerivativesJob < ActiveFedoraIdBasedJob
```
Comment (flyingzumwalt) [inline curation_concerns-models/app/jobs/create_derivatives_job.rb]: Why did you delete this file?  It's in both sufia/pcdm and curation_concerns/master.

File: curation_concerns-models/app/actors/curation_concerns/base_actor.rb (hunk 4 above)
Comment (jpstroop) [inline curation_concerns-models/app/actors/curation_concerns/base_actor.rb]: Cf: https://github.com/projecthydra/sufia/blob/pcdm/sufia-models/app/actors/curation_concern/base_actor.rb#L75-L83

File: curation_concerns-models/app/actors/curation_concerns/base_actor.rb (hunk 3 above)
Comment (jpstroop) [inline curation_concerns-models/app/actors/curation_concerns/base_actor.rb]: Cf: https://github.com/projecthydra/sufia/blob/pcdm/sufia-models/app/actors/curation_concern/base_actor.rb#L60

File: curation_concerns-models/app/jobs/import_url_job.rb [hunk 8]
```diff
@@ -14,7 +14,7 @@ def run
     Tempfile.open(id.gsub('/', '_')) do |f|
       path, mime_type = copy_remote_file(generic_file.import_url, f)
       # attach downloaded file to generic file stubbed out
-      if Sufia::GenericFile::Actor.new(generic_file, user).create_content(f, path, 'content', mime_type)
+      if Sufia::GenericFile::Actor.new(generic_file, user).create_content(f, path, mime_type)
```
Comment (flyingzumwalt) [inline curation_concerns-models/app/jobs/import_url_job.rb]: Should use CurationConcerns::GenericFileActor, not Sufia::GenericFile::Actor

File: curation_concerns-models/app/jobs/ingest_local_file_job.rb [hunk 9]
```diff
@@ -21,7 +20,7 @@ def run
 
     actor = Sufia::GenericFile::Actor.new(generic_file, user)
```
Comment (flyingzumwalt) [inline curation_concerns-models/app/jobs/ingest_local_file_job.rb]: Use `CurationConcerns::GenericFileActor`, not `Sufia::GenericFile::Actor`

File: curation_concerns-models/app/jobs/ingest_local_file_job.rb [hunk 10]
```diff
@@ -21,7 +20,7 @@ def run
...
 
-    if actor.create_content(File.open(path), filename, 'content', mime_type(filename))
+    if actor.create_content(File.open(path), filename, mime_type(filename))
       FileUtils.rm(path)
       Sufia.queue.push(ContentDepositEventJob.new(generic_file.id, user_key))
```
Comment (flyingzumwalt) [inline curation_concerns-models/app/jobs/ingest_local_file_job.rb]: `CurationConcerns.queue`, not `Sufia.queue`

File: curation_concerns-models/app/models/concerns/curation_concerns/file/virus_check.rb [hunk 11]
```diff
@@ -10,9 +10,9 @@ module VirusCheck
       # Default behavior is to raise a validation error and halt the save if a virus is found
       def detect_viruses
         return unless content.changed?
-        Sufia::GenericFile::Actor.virus_check(local_path_for_content)
+        CurationConcerns::GenericFile::Actor.virus_check(local_path_for_content)
```
Comment (flyingzumwalt) [inline curation_concerns-models/app/models/concerns/curation_concerns/file/virus_check.rb]: Use `CurationConcerns::GenericFileActor`, not `CurationConcerns::GenericFile::Actor`

File: curation_concerns-models/app/models/concerns/curation_concerns/model_methods.rb [hunk 12]
```diff
@@ -0,0 +1,20 @@
+module CurationConcerns
+  module ModelMethods
```
Comment (flyingzumwalt) [inline curation_concerns-models/app/models/concerns/curation_concerns/model_methods.rb]: Remove this.  I moved/removed these methods already.  Now they come from the lines where GenericFileBehavior calls:

```ruby
    include Hydra::WithDepositor
    include CurationConcerns::Serializers
```

File: curation_concerns-models/Gemfile (hunk 1 above)
Comment (jpstroop) [inline curation_concerns-models/Gemfile]: There will be a lot of these...I figured I'd just grep for comments and replace them with sed later

File: curation_concerns-models/app/models/single_use_link.rb [hunk 13]
```diff
@@ -0,0 +1,42 @@
+class SingleUseLink < ActiveRecord::Base
```
Comment (flyingzumwalt) [inline curation_concerns-models/app/models/single_use_link.rb]: I think these belong in sufia-core, not curation_concerns.

File: curation_concerns-models/app/models/local_authority_entry.rb [hunk 14]
```diff
@@ -0,0 +1,3 @@
+class LocalAuthorityEntry < ActiveRecord::Base
```
Comment (flyingzumwalt) [inline curation_concerns-models/app/models/local_authority_entry.rb]: I think these belong in sufia-core, not curation_concerns.

File: curation_concerns-models/app/models/subject_local_authority_entry.rb [hunk 15]
```diff
@@ -0,0 +1,2 @@
+class SubjectLocalAuthorityEntry < ActiveRecord::Base
```
Comment (flyingzumwalt) [inline curation_concerns-models/app/models/subject_local_authority_entry.rb]: I think these belong in sufia-core, not curation_concerns.

File: curation_concerns-models/app/services/curation_concerns/full_text_extraction_service.rb [hunk 16]
```diff
@@ -0,0 +1,38 @@
+module Sufia
```
Comment (flyingzumwalt) [inline curation_concerns-models/app/services/curation_concerns/full_text_extraction_service.rb]: `module CurationConcerns`

File: curation_concerns-models/app/services/curation_concerns/generic_file_audit_service.rb [hunk 17]
```diff
@@ -0,0 +1,83 @@
+module Sufia
```
Comment (flyingzumwalt) [inline curation_concerns-models/app/services/curation_concerns/generic_file_audit_service.rb]: `module CurationConcerns`

File: curation_concerns-models/app/services/curation_concerns/generic_file_indexing_service.rb [hunk 18]
```diff
@@ -0,0 +1,14 @@
+module Sufia
```
Comment (flyingzumwalt) [inline curation_concerns-models/app/services/curation_concerns/generic_file_indexing_service.rb]: `module CurationConcerns`

File: curation_concerns-models/app/services/curation_concerns/generic_work_indexing_service.rb [hunk 19]
```diff
@@ -0,0 +1,12 @@
+module Sufia
```
Comment (flyingzumwalt) [inline curation_concerns-models/app/services/curation_concerns/generic_work_indexing_service.rb]: `module CurationConcerns`

File: curation_concerns-models/app/services/curation_concerns/repository_audit_service.rb [hunk 20]
```diff
@@ -0,0 +1,9 @@
+module Sufia
```
Comment (flyingzumwalt) [inline curation_concerns-models/app/services/curation_concerns/repository_audit_service.rb]: `module CurationConcerns`

File: curation_concerns-models/lib/curation_concerns/permissions.rb [hunk 21]
```diff
@@ -0,0 +1,9 @@
+module CurationConcerns
+  module Permissions
```
Comment (flyingzumwalt) [inline curation_concerns-models/lib/curation_concerns/permissions.rb]: I had namespaced these in `CurationConcerns::File::Permissions`.  You're right to put them in `CurationConcerns::File::Permissions`.  Leave these ones, delete `curation_concerns/file/permissions.rb` and `curation_concerns/file/permissions` directory, and update any existing references to `CurationConcerns::File::Permissions` so that they use the new namespace.

Comment (flyingzumwalt) [inline curation_concerns-models/lib/curation_concerns/permissions.rb]: better: move `curation_concerns/file/permissions.rb` to `curation_concerns/permissions.rb` and move `curation_concerns/file/permissions` to `curation_concerns/permissions`.  I think I made a few changes in the modules that are already in `curation_concerns/file/permissions`

**Comment by flyingzumwalt** [comment]
Ok.  Finished code review.

File: curation_concerns-models/app/jobs/batch_update_job.rb (hunk 6 above)
Comment (jpstroop) [inline curation_concerns-models/app/jobs/batch_update_job.rb]: c.f. https://github.com/projecthydra/sufia/blob/pcdm/sufia-models/app/jobs/batch_update_job.rb#L49-L56

**Comment by jpstroop** [comment]
OK @flyingzumwalt, this is back to the single fail we have on master. Merge at will!
